### PR TITLE
mssql select stmt special generation

### LINKED ISSUE #56: CacheRuleInformer creates MySQL specific Select statements
The `CacheRuleInformer#EagerCacheRuleSpecAdapter` currently creates a Select statement which uses the `?` placeholder. This won't work for the other supported databases.

### PR DESCRIPTION
**Pull request by rigazilla**
fix #56 

adding mssql select generation

### CODE REVIEW

File: manager/src/main/java/io/gingersnapproject/k8s/CacheRuleInformer.java [hunk 1]
```diff
@@ -370,17 +359,47 @@ public boolean queryEnabled() {
...
+               .append(" FROM ")
+               .append(eagerRule.getTableName());
+         if (eagerRule.getKey().getKeyColumnsCount() > 0) {
+            sb.append(" WHERE ").append(
+                  whereClause());
+         }
+         selectStmt = sb.toString();
+      }
+
+      private String whereClause() {
```
Comment (ryanemerson) [inline manager/src/main/java/io/gingersnapproject/k8s/CacheRuleInformer.java]: We already have the `Vendor` interface and implementations for DB specific logic.  I think we should add a method to that interface that takes the keys/values as args and produces the select statement. We can then just call `Vendor.fromDbKind` in this adaptor.

File: manager/src/main/java/io/gingersnapproject/database/vendor/Vendor.java [hunk 2]
```diff
@@ -53,4 +54,8 @@ static PrimaryKey extractPrimaryKeyInLowerCase(RowSet<Row> rows) {
       }
       return new PrimaryKey(pkName, pkColumns);
    }
+
+   default String whereClause(List<String> keys) {
```
Comment (ryanemerson) [inline manager/src/main/java/io/gingersnapproject/database/vendor/Vendor.java]: I'm not sure a `default` implementation makes sense given that the placeholders are different for every DB vendor.

Related, could you also add the implementations for Oracle and Postgres? :angel:

Comment (rigazilla) [reply manager/src/main/java/io/gingersnapproject/database/vendor/Vendor.java]: oh sure, I thought mssql was the only one with different placeholder

Comment (wburns) [reply manager/src/main/java/io/gingersnapproject/database/vendor/Vendor.java]: Sadly I already did this in https://github.com/gingersnap-project/cache-manager/pull/49/files#diff-0fd5204bef5db21cd685c49abd1953b797863596f39ca8e8f95fb349e78d2baaR16. I had done it where each vendor only overrides a method to define the parameter with position.

**Comment by wburns** [comment]
Changes look okay to me. Can you double check @ryanemerson ?

**Comment by ryanemerson** [comment]
Thanks @rigazilla

**Pull request merged**